### ISSUE TITLE
Comment & Reference columns in Inbound Shipments need re-thinking 💭

### ISSUE THREAD

**Issue by marthakeezy** [msupply-foundation/open-msupply#5221]
## What went wrong? 😲

Raising after a brief chat with @mariyamsupply 

When viewing Inbound Shipments on Desktop, Browser or Tablet, it is pretty impossible to read the text - even if it is something short, or default like 'From invoice X', the text gets horribly cut. My concern is on tablet, you have to hold down the cell to read the full text and your finger has to be positioned in a very specific way to see said text. Also, it adds clutter to the screen and serves minimal purpose.

<img width="492" alt="Screenshot 2024-10-24 at 4 07 19 PM" src="https://github.com/user-attachments/assets/f3d9ed52-2bc2-4fd0-b269-8e850e793858">

This is how I find myself using tablet most of the times: 

<img width="910" alt="Screenshot 2024-10-24 at 4 09 06 PM" src="https://github.com/user-attachments/assets/63ee33a4-ea6c-4a73-a81d-b541fe308a23">

The pathway to finding that information within Inbound Shipments also does not seem super easy to me -> user is required to click on the inbound shipment -> then on 'More' to see the comment. With how busy the page is, it might also not be super obvious to them where to look for that information. 

<img width="1091" alt="Screenshot 2024-10-24 at 4 10 59 PM" src="https://github.com/user-attachments/assets/f0d3f4b0-3292-437b-97ec-f9580d5e69d4">

## Expected behaviour 🤔

I enjoy the way we solved this problem when viewing breaches related to cold chain. There is a simple icon indicating text present, which is easy to click on when using tablets and computers. Not every Inbound Shipment has reference or comment attached, so for those that do, can we have this icon shown instead of the text that is hard to read?

https://github.com/user-attachments/assets/f314399f-a60d-4c8c-99b8-9b19d2c0730d

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Go to Replenishment -> Inbound Shipment
2. Look at the page from a tablet and a desktop
3. See the text that is hard to read
4. Let's think of better ways to show this!

## Your environment 🌱

- Open mSupply Version: `V2.3.1-RC1`
- Legacy mSupply Central Server Version: `V7.16.8`

- Platform: `Browser/Desktop/Tablet`

- Database type: `SQLite2`

**Comment by marthakeezy** [comment]
Maybe this needs to be applied to all areas where the text gets cut off anyway?

- Outbound Shipment (mild cutting happening)
- Requisitions (mild cutting)
- Customer Returns (intense cutting)
- Inbound Shipment (intense cutting)
- Internal Orders (mild cutting)
- Supplier Returns (intense cutting)
- Prescriptions (moderate cutting)
- Monitoring @ Cold chain - chef's kiss!

**Comment by andreievg** [comment]
Triage: Changing to low priority, very low impact. Since we need a general way to display more data in tables on tablet, should be considered in that context. Also need to decide what the list view is for, is it to see detailed information about the entry (like comment) or just the stores/names and statuses

**Comment by mariyamsupply** [comment]
Potential solution is to have a comment icon for the lines that have comments and show a full comment on click. Will spare some space as well 
<img width="1024" alt="comments" src="https://github.com/user-attachments/assets/9c97e125-4378-48b6-815b-bf52e86bebc2">

<img width="1024" alt="comments to see" src="https://github.com/user-attachments/assets/20d24152-3c74-4ae7-bf48-62baf9b1e14b">

**Comment by roxy-dao** [comment]
> I'd suggest implementing a comment column and making the reference column wider

Why not have reference as a comment icon as well since some might be long (specially for transfers?)

**Comment by mark-prins** [comment]
personally - I think reference is better as text in the column - to have it readable. The comment is usually a longer / additional note and so that's fine being hidden. 

agree that the generated reference is a little too wordy - but the intention with reference is a short piece of text
